### PR TITLE
Added DuplicateKeyWithUniqueIndexException for SQL code 2601

### PR DESCRIPTION
**Pull request by sasestil**

### CODE REVIEW

File: EFxceptions.Identity.Tests/EFxceptionServiceTests.cs [hunk 1]
```diff
@@ -112,6 +112,27 @@ public void ShouldThrowForeignKeyConstraintConflictException()
...
+            string randomErrorMessage = new MnemonicString().GetValue();
+            SqlException duplicateKeySqlException = CreateSqlException();
+
+            var dbUpdateException = new DbUpdateException(
+                message: randomErrorMessage,
+                innerException: duplicateKeySqlException);
+
+            this.sqlErrorBrokerMock.Setup(broker =>
+                    broker.GetSqlErrorCode(duplicateKeySqlException))
+                .Returns(sqlDuplicateKeyErrorCode);
```
Comment (hassanhabib) [inline EFxceptions.Identity.Tests/EFxceptionServiceTests.cs:129]: Push this line to be one tab ahead of the one above it please.

File: EFxceptions.Tests/Services/EFxceptionServiceTests.cs [hunk 2]
```diff
@@ -112,6 +112,27 @@ public void ShouldThrowForeignKeyConstraintConflictException()
...
+            string randomErrorMessage = new MnemonicString().GetValue();
+            SqlException duplicateKeySqlException = CreateSqlException();
+
+            var dbUpdateException = new DbUpdateException(
+                message: randomErrorMessage,
+                innerException: duplicateKeySqlException);
+
+            this.sqlErrorBrokerMock.Setup(broker =>
+                    broker.GetSqlErrorCode(duplicateKeySqlException))
+                .Returns(sqlDuplicateKeyErrorCode);
```
Comment (hassanhabib) [inline EFxceptions.Tests/Services/EFxceptionServiceTests.cs:129]: Same as above.

**Comment by sasestil** [comment]
@hassanhabib 

Hi,
I have dilemma about exception name. Is this name "DuplicateKeyUniqueIndexException" is ok or will be better to be "DuplicateKeyWithUniqueIndexException"

**Comment by hassanhabib** [comment]
> @hassanhabib 
> 
> Hi,
> I have dilemma about exception name. Is this name "DuplicateKeyUniqueIndexException" is ok or will be better to be "DuplicateKeyWithUniqueIndexException"

I like the latter more.

**Comment by sasestil** [comment]
@hassanhabib 
Can you please review and marge this pull request.

Thank you.

**Review by hassanhabib — APPROVED** [review]

**Pull request merged**